### PR TITLE
Use PHP's DateTime::ATOM for true ISO-8601 support

### PR DESCRIPTION
**Pull request by arnested**
Currently we use PHP's `DateTime::ISO8601` for the date-time properties but according to http://php.net/manual/en/class.datetime.php#datetime.constants.iso8601 it is actually not compatible with ISO-8601.

Instead we should use `DateTime::ATOM`.

### CODE REVIEW

**Comment by wing328** [comment]
@arnested thanks for the PR. I don't see change in the object serializer mustache template. I only found changes to the PHP Petstore sample.

Would you please take a look and rebase on the latest master?

**Comment by arnested** [comment]
Yeah, I apparently got something wrong there. Rebasing and fixing right now. New push in a minute.

Sorry for the trouble, @wing328.

**Comment by wing328** [comment]
@arnested no worry. Please take your time.

**Comment by arnested** [comment]
Just a minor glitch -- fixed and push now.

File: samples/client/petstore/php/SwaggerClient-php/README.md [hunk 1]
```diff
@@ -46,91 +46,91 @@ All URIs are relative to *http://petstore.swagger.io/v2*
...
-*StoreApi* | [**placeOrder**](docs/StoreApi.md#placeorder) | **POST** /store/order | Place an order for a pet
-*UserApi* | [**createUser**](docs/UserApi.md#createuser) | **POST** /user | Create user
-*UserApi* | [**createUsersWithArrayInput**](docs/UserApi.md#createuserswitharrayinput) | **POST** /user/createWithArray | Creates list of users with given input array
-*UserApi* | [**createUsersWithListInput**](docs/UserApi.md#createuserswithlistinput) | **POST** /user/createWithList | Creates list of users with given input array
-*UserApi* | [**deleteUser**](docs/UserApi.md#deleteuser) | **DELETE** /user/{username} | Delete user
-*UserApi* | [**getUserByName**](docs/UserApi.md#getuserbyname) | **GET** /user/{username} | Get user by user name
-*UserApi* | [**loginUser**](docs/UserApi.md#loginuser) | **GET** /user/login | Logs user into the system
-*UserApi* | [**logoutUser**](docs/UserApi.md#logoutuser) | **GET** /user/logout | Logs out current logged in user session
-*UserApi* | [**updateUser**](docs/UserApi.md#updateuser) | **PUT** /user/{username} | Updated user
+*PetApi* | [**addPet**](PetApi.md#) | **POST** /pet | Add a new pet to the store
```
Comment (wing328) [inline samples/client/petstore/php/SwaggerClient-php/README.md]: Are you using a customized version of `PhpClientCodegen` and `DefaultCodegen` somehow? e.g.  {{operationIdLowerCase}} seems not defined.

Comment (arnested) [inline samples/client/petstore/php/SwaggerClient-php/README.md]: After rebasing I forgot to `mvn clean package` so this was probably generated with an older / somehow other version of codegen.

I've rebuild codege, regenerated the petstore sample, and run the phpunit test suite now.

CI is currently building.

Sorry for the noise again and thank you for catching it.

**Comment by wing328** [comment]
**Upgrade note from 2.1.5 to 2.1.6**

Starting from 2.1.6, `DateTime::ATOM` will be used instead of `DateTime::ISO8601`. The change is based on the 


> Note: This (DateTime::ISO8601) format is not compatible with ISO-8601, but is left this way for backward compatibility reasons. Use DateTime::ATOM or DATE_ATOM for compatibility with ISO-8601 instead.

Ref: http://php.net/manual/en/class.datetime.php#datetime.constants.iso8601

If this breaks your application after upgrade, please reply to let us know and we'll share a workaround.

**Pull request merged**